### PR TITLE
Add a way to Scan a QR Code that contain a Gziped JSON

### PR DESCRIPTION
**Pull request by rjmangubat23**
### Issue

#21 Add a way to Scan a QR Code that contain a Gziped JSON 

### Changes

- Add Scan a QR Code containing Gzipped JSON implementation 
- Add Gzipped JsonPath dependencies
- Refactor and code clean up of library and app core classes
- Update smartscanner-android api via https://github.com/idpass/smartscanner-android/pull/3

### CODE REVIEW

File: core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt [hunk 1]
```diff
@@ -109,16 +117,37 @@ class BarcodeAnalyzer(
         }
     }
 
-    private fun sendAnalyzerResult(activity: Activity, result: String? = null) {
+    private fun sendGzippedResult(rawValue : String, rawBytes: ByteArray) {
```
Comment (jeremi) [inline core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt:120]: It is not always gzipped, can you rename to `sendQrCodeResult` maybe?

File: core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt [hunk 2]
```diff
@@ -109,16 +117,37 @@ class BarcodeAnalyzer(
...
+        // parse and read gzip json and add to bundle intent
+        val bundle = Bundle()
+        val isGzipped = intent.getBooleanExtra(ScannerConstants.GZIPPED_ENABLED, true)
+        val isJson = intent.getBooleanExtra(ScannerConstants.JSON_ENABLED, true)
+        val jsonPath = intent.getStringExtra(ScannerConstants.JSON_PATH)
+        // check gzipped parameters for bundle return result
+        if (isGzipped) {
+            val gzippedData = GzipUtils.decompress(rawBytes)
+            jsonPath?.let { path ->
```
Comment (jeremi) [inline core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt:129]: What if the json is not gzipped? JSON will not always be gzipped

File: core-lib/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -16,9 +16,12 @@
             android:configChanges="orientation|keyboardHidden"
             android:screenOrientation="portrait">
             <intent-filter>
+                <!--Bundle intent-->
                 <action android:name="org.idpass.smartscanner.MRZ_SCAN" />
                 <action android:name="org.idpass.smartscanner.BARCODE_SCAN" />
                 <action android:name="org.idpass.smartscanner.IDPASS_LITE_SCAN" />
+                <action android:name="org.idpass.smartscanner.QRCODE_SCAN" />
```
Comment (jeremi) [inline core-lib/src/main/AndroidManifest.xml:23]: we need `org.idpass.smartscanner.QRCODE_SCAN` and `org.idpass.smartscanner.odk.QRCODE_SCAN` for consistency

File: core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt [hunk 4]
```diff
@@ -109,16 +117,37 @@ class BarcodeAnalyzer(
         }
     }
 
-    private fun sendAnalyzerResult(activity: Activity, result: String? = null) {
+    private fun sendGzippedResult(rawValue : String, rawBytes: ByteArray) {
+        // parse and read gzip json and add to bundle intent
+        val bundle = Bundle()
+        val isGzipped = intent.getBooleanExtra(ScannerConstants.GZIPPED_ENABLED, true)
```
Comment (jeremi) [inline core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt:123]: default to false

File: core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt [hunk 5]
```diff
@@ -109,16 +117,37 @@ class BarcodeAnalyzer(
...
+        val isJson = intent.getBooleanExtra(ScannerConstants.JSON_ENABLED, true)
```
Comment (jeremi) [inline core-lib/src/main/java/org/idpass/smartscanner/lib/barcode/BarcodeAnalyzer.kt:124]: default to false

**Review by jeremi — CHANGES_REQUESTED** [review]

**Comment by rjmangubat23** [comment]
@jeremi requested changes are now resolved in latest commit. May I ask for re-review and approval so that I can be able to merge? Thank you!

**Pull request merged**